resource/schema/bool_attribute.go

```diff
@@ -135,6 +138,12 @@ type BoolAttribute struct {
 	//
 	// Any errors will prevent further execution of this sequence or modifiers.
 	PlanModifiers []planmodifier.Bool
+
+	// Default defines a default value for the attribute. The default value
+	// handling occurs during the `PlanResourceChange` RPC and modifies the
+	// Terraform-provided proposed new state before the framework performs
+	// its check between the proposed new state and prior state.
```

Comment:
Some additional tidbits likely worth mentioning to provider developers:

- Default defines a _proposed new state (plan)_ value for the attribute _if the configuration value is null_ 😄 
- Default prevents the framework from automatically marking the value as unknown during planning when other proposed new state changes are detected.

🤔 If the computed attribute value _could_ change due to other changes, provider developers either need a way out of that situation with a framework-defined plan modifier to forcibly mark as unknown (e.g. #605) or the documentation should say that defaults may not be for you. I think I prefer the documentation saying to avoid defaults in that situation, but curious if you have any thoughts.

Reply:
Have amended the comments. I agree about the guidance relating to avoiding defaults if computed values could change due to other changes and have added words to that effect.